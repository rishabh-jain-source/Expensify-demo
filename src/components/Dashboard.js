import React from 'react'
import ConnectedExpenseList from './ExpenseList'
import ConnectedExpenseListFilters from './ExpenseListFilters'
import  ConnectedExpensifySummary  from './ExpensifySummary'
const Dashboard = () => (
    <div>
        <h1>Dashboard</h1>
        <ConnectedExpensifySummary/>
        <ConnectedExpenseListFilters/>
        <ConnectedExpenseList/>
    </div>
)
export default Dashboard
    