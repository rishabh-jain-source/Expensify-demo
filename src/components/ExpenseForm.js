import React,{Fragment} from 'react'
import moment from 'moment'
import { SingleDatePicker } from 'react-dates'

class ExpenseForm extends React.Component{
    constructor(props) {
        super(props);
        this.state = {
            description: props.expense?props.expense.description: '',
            amount: props.expense?props.expense.amount:'',
            note: props.expense?props.expense.note:'',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calenderFocused: false,
            error:''
        }
    }
    
    onDescriptionChange = (e) => {
        const description = e.target.value
        
        this.setState(() => {
            return {
                description
                
            }
        })
    }
    onAmountChange = (e) => {
        const amount = e.target.value
        
            this.setState(() => {
                return {
                    amount
                   
                }
                
            })
        
    }
    onNoteChange = (e) => {
        const note = e.target.value
        this.setState(() => {
            return {
                note
            }
        })
    }
    onDateChange = (createdAt) => {
        if (createdAt){
            this.setState( () => ({ createdAt }) );
        }
    };
    onFocusChange = ({ focused }) => {
        this.setState(() => {
            return {
                calenderFocused:focused
            }
        })
    }
    formSubmit = (e) => {
        e.preventDefault()
        if (!this.state.description || !this.state.amount) {
            if (!this.state.description) {
                this.setState(() => {
                    return {
                        error:'Enter Description*'
                    }
                })
            }
            else {
                this.setState(() => {
                    return {
                        error:'Enter Amount*'
                    }
                })
            }
        }
        else {
            this.setState(() => {
                return {
                    error:''
                }
            })
            console.log("submitted")
            this.props.onSubmit({
                description: this.state.description,
                note: this.state.note,
                createdAt: `${this.state.createdAt}`,
                amount:parseFloat(this.state.amount,10)
            })
        }
    }
    render() {
        return (<div >
            
                <form onSubmit={this.formSubmit} className="form" >
                <input className="text-input input-group" type="text"  placeholder="Description" value={this.state.description} onChange={this.onDescriptionChange}></input>
                {this.state.error && <p className="error">{ this.state.error  }</p>}
            <input className="text-input input-group" type="number" placeholder="Amount" value={this.state.amount} onChange={this.onAmountChange}></input>
            <textarea className="textarea-input input-group" placeholder="Note Optional" value={this.state.note} onChange={this.onNoteChange}></textarea>
            <SingleDatePicker
                date={this.state.createdAt}
                    onDateChange={() => { this.onDateChange(date) }}
                focused={this.state.calenderFocused}
                onFocusChange={this.onFocusChange}
                numberOfMonths={1}
                isOutsideRange={()=>{false}}
            />
            </form>
                <button className="button "  onClick={this.formSubmit} > Save</button>
            
        </div>
        )
    }
}

export default ExpenseForm