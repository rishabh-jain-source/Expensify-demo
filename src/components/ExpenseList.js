import React from 'react'
import { connect } from 'react-redux'
import ConnectedExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/getVisible'
export const ExpenseList = (props) => (
    <div className="content-container">
        <h1 className=".page-header__title">Expense List</h1>
        <h1 className="table-length">{props.expenses.length}</h1>
       
        {props.expenses.map((expense) => {
            return <ConnectedExpenseListItem key={expense.id}  {...expense}/>
        })}
    </div>
)
const mapPropsTostate = (state) => {
    return {
        expenses: getVisibleExpenses(state.expenses,state.filters)
        
    }
}
const ConnectedExpenseList=connect(mapPropsTostate)(ExpenseList)
export default ConnectedExpenseList