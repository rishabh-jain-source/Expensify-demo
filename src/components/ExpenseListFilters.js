import React from 'react'
import { connect } from 'react-redux'
import { removeExpense } from '../actions/expenses'
import { setTextFilter, sortByDate,sortByAmount } from '../actions/filters'

const ExpenseListFilters = (props) => (
    <div className="content-container">
        <div className="input-group">
            <div className="input-group__item">
                <input type="text"
                    className="text-input"
                    value={props.filters.text} onChange={
                    (e) => {
                    props.dispatch(setTextFilter(e.target.value))
                    console.log(e.target.value)
                    }
                }
                />
            </div>
            <div className="input-group__item">
                <select className="text-input" value={props.filters.sortBy} onChange={
                    (e) => {
                        if (e.target.value === 'date') {
                            props.dispatch(sortByDate())
                        }
                        else if (e.target.value === 'amount') {
                            props.dispatch(sortByAmount())
                        }
                    }
                }>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                 </select>
            </div>
        </div>   
    </div>
)
const mapPropsTostate = (state) => {
    return {
        filters: state.filters
    }
}
const ConnectedExpenseListFilters=connect(mapPropsTostate)(ExpenseListFilters)
export default ConnectedExpenseListFilters
