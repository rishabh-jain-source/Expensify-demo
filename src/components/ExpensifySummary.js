import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import getVisibleExpenses from './../selectors/getVisible'
import ExpensesTotal from '../selectors/ExpensesTotal'
export const ExpensifySummary = ({ expenseCount, expenseTotal }) => {
    const expenseWord = (expenseCount > 1 ? 'expenses' : 'expense')
    const expenseFormat=numeral(expenseTotal).format('$0,000.00')
    return (
        <div>
            <h1>viewing {expenseCount} {expenseWord} with total of { expenseFormat}</h1>
        </div >
    )
}
const mapStateToProps=(state)=> {
    const visible = getVisibleExpenses(state.expenses, state.filters)
    return {
        expenseCount: visible.length,
        expenseTotal: ExpensesTotal(visible)
    }
}

 export default connect(mapStateToProps)(ExpensifySummary)   
