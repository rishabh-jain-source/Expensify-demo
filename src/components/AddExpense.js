// import React from 'react'
// import { connect } from 'react-redux'
// import { addExpense, startAddExpense } from '../actions/expenses'
// import ExpenseForm from './ExpenseForm'


// const AddExpense = (props) => (
//     <div>
//         <h1>AddExpense</h1>
//         <ExpenseForm onSubmit={(expense) => {
//             console.log('My expense'+ JSON.stringify(expense))
//              props.dispatch(addExpense(expense))
//              props.history.push('/dashboard ')
//             //console.log(expense);
//         } }/>
//     </div>
// )
// const mapDispatchToProps = (dispatch) => ({
//   startAddExpense:(expense)=> dispatch(startAddExpense(expense))   
// })
// export default connect()(AddExpense)
    
// // import React from "react";
// // import { connect } from 'react-redux';
// // import ExpenseForm from './ExpenseForm';
// // import { addExpense,startAddExpense } from '../actions/expenses';


// // export class AddExpense extends React.Component {

// // function myWorld(){}


// //     onSubmit = (expense) => {
// //         console.log("My expense Data"+ expense)
// //         this.addExpense(expense);
// //         this.props.history.push('/');
// //         this.myWorld()
// //     };
// //     render(){
// //         return (
// //             <div>
// //                 <h1>Add Expense</h1>
// //                 <ExpenseForm 
// //                     onSubmit = {this.onSubmit}
// //                 />
// //             </div>
// //         );
// //     }
// // }

// // const mapDispatchToProps = (dispatch) => ({
// //     startAddExpense: (expense) => dispatch(startAddExpense(expense))
// // });

// //export default connect(undefined, mapDispatchToProps)(AddExpense);

import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export class AddExpense extends React.Component {
  onSubmit = (expense) => {
      
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
       
        <div className="content-container"><h1>Add Expense</h1>
          <div className="page-header__title">
          </div>
          </div>
          </div>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpense);