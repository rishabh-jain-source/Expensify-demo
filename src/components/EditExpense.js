import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from "./ExpenseForm";
import { editExpense ,startRemoveExpense} from '../actions/expenses'


class EditExpense extends React.Component{//= (props) => (
    removeExp = () => {
        //onsole.log(this.props.expense.id)
        this.props.startRemoveExpense({ id: this.props.expense.id })
        this.props.history.push('/')
        
    }
    render() {
        return(
        <div>
        
        
            <h1>EditExpense</h1>
            <h1>{this.props.match.params.id}</h1>
            <ExpenseForm
            
                expense={this.props.expense}
                onSubmit={(expense) => {
                    this.props.dispatch(editExpense(this.props.expense.id, expense))
                    
                }} />
            <button onClick={this.removeExp()}>remove Expense</button>
            </div>
        )
    }
}


const mapStateToProps = (state,props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id===props.match.params.id
        }),
    }
}
const mapDispatchToProps = (dispatch, props) => {
    return {
        editExpense:(id,expense)=>dispatch(editExpense(id,expense)),
        startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
    }
}
export default connect(mapStateToProps,mapDispatchToProps)(EditExpense)