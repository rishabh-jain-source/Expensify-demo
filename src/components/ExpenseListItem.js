import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { removeExpense } from '../actions/expenses'
import EditExpense from './EditExpense'
import { Route } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

export const ExpenseListItem = ({ id,dispatch,description, amount, createdAt }) => (
    <div>
        {console.log(id)}
        <Link to={`editExpense/${id}`}>{description}</Link>
        <p>
            {numeral(amount).format('$0,00.00')}
        -
        {moment(createdAt).format('MMMM Do,YYYY')}</p>
        <p>{ id}</p>
        {/* <button onClick={() => {
            dispatch(removeExpense({id}))
        }}>Remove</button> */}
    </div>
)

const ConnectedExpenseListItem=connect()(ExpenseListItem)
export default ConnectedExpenseListItem
    
