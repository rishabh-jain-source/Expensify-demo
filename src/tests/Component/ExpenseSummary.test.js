import { shallow } from 'enzyme'
import React from 'react'
import {ExpensifySummary} from '../../components/ExpensifySummary'

test('SHould render expensifySummary', () => {
    const wrapper = shallow(<ExpensifySummary expenseTotal={100} expenseCount={1} />)
    expect(wrapper).toMatchSnapshot();
})
test('SHould render multiple expensifySummary', () => {
    const wrapper = shallow(<ExpensifySummary expenseTotal={100787} expenseCount={10} />)
    expect(wrapper).toMatchSnapshot();
})