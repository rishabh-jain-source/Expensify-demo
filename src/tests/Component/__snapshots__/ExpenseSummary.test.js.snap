// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHould render expensifySummary 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensifySummary
    expenseCount={1}
    expenseTotal={100}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        viewing 
        1
         
        expense
         with total of 
        100
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "viewing ",
          1,
          " ",
          "expense",
          " with total of ",
          100,
        ],
      },
      "ref": null,
      "rendered": Array [
        "viewing ",
        1,
        " ",
        "expense",
        " with total of ",
        100,
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          viewing 
          1
           
          expense
           with total of 
          100
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "viewing ",
            1,
            " ",
            "expense",
            " with total of ",
            100,
          ],
        },
        "ref": null,
        "rendered": Array [
          "viewing ",
          1,
          " ",
          "expense",
          " with total of ",
          100,
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`SHould render multiple expensifySummary 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensifySummary
    expenseCount={10}
    expenseTotal={100787}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        viewing 
        10
         
        expenses
         with total of 
        100787
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "viewing ",
          10,
          " ",
          "expenses",
          " with total of ",
          100787,
        ],
      },
      "ref": null,
      "rendered": Array [
        "viewing ",
        10,
        " ",
        "expenses",
        " with total of ",
        100787,
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          viewing 
          10
           
          expenses
           with total of 
          100787
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "viewing ",
            10,
            " ",
            "expenses",
            " with total of ",
            100787,
          ],
        },
        "ref": null,
        "rendered": Array [
          "viewing ",
          10,
          " ",
          "expenses",
          " with total of ",
          100787,
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
